CDP4CrossViewEditor/ViewModels/ThingSelectorViewModel.cs

```diff
@@ -0,0 +1,240 @@
+        /// </summary>
+        public ReactiveList<Thing> SourceThingList
+        {
+            get => this.sourceThingList;
+            private set => this.RaiseAndSetIfChanged(ref this.sourceThingList, value);
+        }
+
+        /// <summary>
+        /// Gets or sets source list(element/parameter list)
+        /// </summary>
+        public ReactiveList<Thing> TargetThingList
+        {
+            get => this.targetThingList;
+            private set => this.RaiseAndSetIfChanged(ref this.targetThingList, value);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ThingSelectorViewModel"/> class.
+        /// </summary>
+        /// <param name="iteration">Current opened iteration <see cref="Iteration"/> </param>
+        /// <param name="thingClassKind">Current type of thing <see cref="ClassKind"/></param>
+        public ThingSelectorViewModel(Iteration iteration, ClassKind thingClassKind)
+        {
+            this.Iteration = iteration;
+            this.ThingClassKind = thingClassKind;
+
+            this.SourceThingList = new ReactiveList<Thing>
+            {
+                ChangeTrackingEnabled = true
+            };
+
+            this.TargetThingList = new ReactiveList<Thing>
+            {
+                ChangeTrackingEnabled = true
+            };
+
+            this.MoveItemsToSource = ReactiveCommand.Create();
+            this.MoveItemsToSource.Subscribe(_ => this.ExecuteMoveToSource());
+
+            this.MoveItemsToTarget = ReactiveCommand.Create();
+            this.MoveItemsToTarget.Subscribe(_ => this.ExecuteMoveToTarget());
+
+            this.MoveItemsUp = ReactiveCommand.Create();
+            this.MoveItemsUp.Subscribe(_ => this.ExecuteMoveUp());
+
+            this.MoveItemsDown = ReactiveCommand.Create();
+            this.MoveItemsDown.Subscribe(_ => this.ExecuteMoveItemsDown());
+
+            this.ClearItems = ReactiveCommand.Create();
+            this.ClearItems.Subscribe(_ => this.ExecuteClearItems());
+
+            this.SortItems = ReactiveCommand.Create();
+            this.SortItems.Subscribe(_ => this.ExecuteSortItems());
+        }
+
+        /// <summary>
+        /// Bind source data depends on thing type <see cref="ClassKind"/>
+        /// </summary>
+        public void BindData()
+        {
+            switch (this.ThingClassKind)
+            {
+                case ClassKind.ElementBase:
+                    this.BindElements();
+                    break;
+
+                case ClassKind.ParameterBase:
+                    this.BindParameters();
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Executes move to source command <see cref="MoveItemsToSource"/>
+        /// Things will be moved back to selection source
+        /// </summary>
+        private void ExecuteMoveToSource()
+        {
+            // Deliberate empty, the method behaviour will be defined shortly
```

Comment:
It might be worth it to put links to the business logic implementation task here.